WcaOnRails/app/controllers/server_status_controller.rb

```diff
@@ -2,6 +2,22 @@
 class ServerStatusController < ApplicationController
   MINUTES_IN_WHICH_A_JOB_SHOULD_HAVE_STARTED_RUNNING = 5
 
+  def self.bad_keys_by_type_by_locale
+    @bad_keys_by_type_by_locale ||= begin
+      bad_keys_by_type_by_locale = {}
```

Comment:
Good idea!

The problem with this is that `BAD_KEYS.keys` starts out empty, which would break our [`bad_keys` computation further down in the file](https://github.com/jfly/worldcubeassociation.org/blob/ca011bcdcbe29690ab218bdc4b12b20312ef52e7/WcaOnRails/app/controllers/server_status_controller.rb#L26).

It turns out that Rails has a nifty `each_with_object` that is nice here:

```ruby
BAD_KEYS_BY_TYPE_BY_LOCALE = (I18n.available_locales - [:en]).each_with_object({}) do |locale, hash|
  ref_english = Locale.new('en')
  missing, unused, outdated = Locale.new(locale, true).compare_to(ref_english)
  hash[locale] = { missing: missing, unused: unused, outdated: outdated }
end
```

Or if we don't want the dependency on RoR, Ruby [has a few native solutions](http://stackoverflow.com/questions/2943422/in-ruby-how-do-i-make-a-hash-from-an-array) for building up hashes:

```ruby
BAD_KEYS_BY_TYPE_BY_LOCALE = Hash[(I18n.available_locales - [:en]).map do |locale|
  ref_english = Locale.new('en')
  missing, unused, outdated = Locale.new(locale, true).compare_to(ref_english)
  bad_keys_by_type = { missing: missing, unused: unused, outdated: outdated }
  [ locale, bad_keys_by_type ]
end]
```